git-annex-remote-rclone

```
@@ -16,268 +14,175 @@
-
-			getconfig target
-			REMOTE_TARGET="$RET"
-
-			getconfig rclone_layout
-			RCLONE_LAYOUT="$RET"
-			validate_layout
-
-			echo PREPARE-SUCCESS
-		;;
-		TRANSFER)
-			op="$2"
-			key="$3"
-			shift 3
-			file="$*"
-			case "$op" in
-				STORE)
-					# Store the file to a location
-					# based on the key.
-					# XXX when at all possible, send PROGRESS
-					calclocation "$key"
-					if [ ! -e "$file" ]; then
-						echo TRANSFER-FAILURE STORE "$key" "asked to store non-existent file $file"
-					else
-						if runcmd rclone copy "$file" "$LOC"; then
-							echo TRANSFER-SUCCESS STORE "$key"
-						else
-							echo TRANSFER-FAILURE STORE "$key"
-						fi
-					fi
-				;;
-				RETRIEVE)
-					# Retrieve from a location based on
-					# the key, outputting to the file.
-					# XXX when easy to do, send PROGRESS
-					calclocation "$key"
-					# http://stackoverflow.com/questions/31396985/why-is-mktemp-on-os-x-broken-with-a-command-that-worked-on-linux
-					if GA_RC_TEMP_DIR=$(mktemp -d "${TMPDIR:-/tmp}/rclone-annex-tmp.XXXXXXXXX") &&
-						runcmd rclone copy "$LOC$key" "$GA_RC_TEMP_DIR" &&
-						mv "$GA_RC_TEMP_DIR/$key" "$file" &&
-						rmdir "$GA_RC_TEMP_DIR"; then
-						echo TRANSFER-SUCCESS RETRIEVE "$key"
-					else
-						echo TRANSFER-FAILURE RETRIEVE "$key"
-					fi
-				;;
-			esac
-		;;
-		CHECKPRESENT)
-			key="$2"
-			calclocation "$key"
-
-			do_checkpresent "$key" "$LOC$key"
-		;;
-		REMOVE)
-			key="$2"
-			calclocation "$key"
-
-			do_remove "$key" "$LOC$key"
-		;;
-		*)
-			# The requests listed above are all the ones
-			# that are required to be supported, so it's fine
-			# to say that any other request is unsupported.
-			echo UNSUPPORTED-REQUEST
-		;;
-	esac
-done
-
 
-# XXX anything that needs to be done at shutdown can be done here
+import os
+import re
+import sys
+import shutil
+import tempfile
+import subprocess
+
+def say(cmd, *messages):
+    print(cmd, *messages)
+    sys.stdout.flush()
+
+def ask(cmd, *query):
```

This should return the empty string by default.

---

Is this referring to a situation in which a non `VALUE` key is sent as response? I'd rather throw an error as an empty string could lead the caller into false logic.

---

The bash version returns an empty string if the response does not start with "`VALUE`".  The Python version should do *something* in that case.